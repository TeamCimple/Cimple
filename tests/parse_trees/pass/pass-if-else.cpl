int main() {
	int i;
	
	i = 1;
	if (i = 1) {
		printf("Hello");
	} else if (i = 2) {
		printf("Hello Again");
	} else if (i = 3) {
		printf("Hello Hello Again");
	} else {
		printf("Bye");
	}
	return i;
}
