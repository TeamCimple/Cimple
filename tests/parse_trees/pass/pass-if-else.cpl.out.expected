FuncDecl(Name: Var(main) ReturnType: int Parameters:  Body: CompoundStatement(Declarations: (int Var(i),),  StatementList: Statement(EQUALS(Identifier(i), Int(1))), IF EQUALS(Identifier(i), Int(1)) CompoundStatement(Declarations:  StatementList: Statement(Call(Receiver()FunctionName: Identifier(printf) Params: String("Hello"))))
 IF EQUALS(Identifier(i), Int(2)) CompoundStatement(Declarations:  StatementList: Statement(Call(Receiver()FunctionName: Identifier(printf) Params: String("Hello Again"))))
 IF EQUALS(Identifier(i), Int(3)) CompoundStatement(Declarations:  StatementList: Statement(Call(Receiver()FunctionName: Identifier(printf) Params: String("Hello Hello Again"))))
 CompoundStatement(Declarations:  StatementList: Statement(Call(Receiver()FunctionName: Identifier(printf) Params: String("Bye")))), RETURN(Identifier(i)))) 
